docs/_docs/02_testing-best-practices/testing-plan-output.md

```diff
@@ -0,0 +1,40 @@
+---
+layout: collection-browser-doc
+title: Testing plan output
+category: testing-best-practices
+excerpt: >-
+  Testing specific changes described in Terraform plan output
+tags: ["testing-best-practices"]
+order: 202
+nav_title: Documentation
+nav_title_link: /docs/
+---
+
+Sometimes you can get the information you need in your tests from a `terraform plan`; you may not need to do an `apply`.  Terratest provides a way to use certain information from a `terraform plan` in your tests.
+
+The `InitAndPlanWithInfo` function will return a `PlanInfo` struct that will contain information about which resources would be changed and what specifically about them would change.
+
+For example, let's say you have some `web_server` resources that you're creating with [`for_each`](https://www.terraform.io/docs/configuration/resources.html#for_each-multiple-resource-instances-defined-by-a-map-or-set-of-strings), and you're using the `for_each` values in the name:
+
+```hcl
+resource "aws_instance" "web_server" {
+  for_each = ["env1", "env2", "env3"]
+  name = "web_server_${each.value}"
+  ...
+}
+
+if you want to assert that the names of `web_server` resources that are definied with a terraform `for_each`, you can test this by making assertions against info in this `PlanInfo` struct:
```

Comment:
It's probably a good idea to show the actual `plan` output here and then explain how that is parsed into a `struct`. You can then show side by side how to read certain aspects of the plan from the struct and do checks on them.

A popular example to include here may be how to read security group settings from the `plan` and ensure there are no CIDR blocks set to `0.0.0.0/0`.